install.sh

```diff
@@ -0,0 +1,369 @@
+fi
+
+if [[ ! -w "$INSTALL_PATH" ]]; then
+  abort "No write permission to $INSTALL_PATH. Try running with sudo or specify a different directory with --dir."
+fi
+
+TMP_DIR="$(mktemp -d)"
+cleanup() {
+  rm -rf "$TMP_DIR"
+}
+trap cleanup EXIT
+
+if [[ "$VERSION" == "latest" ]]; then
+  ohai "Fetching latest release information..."
+  RELEASE_URL="https://api.github.com/repos/agentuity/cli/releases/latest"
+  VERSION=$(curl -s $RELEASE_URL | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+  if [[ -z "$VERSION" ]]; then
+    abort "Failed to fetch latest version information"
+  fi
+fi
+
+VERSION="${VERSION#v}"
+
+if [[ "$OS" == "Windows" ]]; then
+  if [[ "$ARCH" == "x86_64" ]]; then
+    DOWNLOAD_FILENAME="agentuity-x64.msi"
+  elif [[ "$ARCH" == "arm64" ]]; then
+    DOWNLOAD_FILENAME="agentuity-arm64.msi"
+  else
+    DOWNLOAD_FILENAME="agentuity-x86.msi"
+  fi
+else
+  DOWNLOAD_FILENAME="agentuity_${OS}_${ARCH}.${EXTENSION}"
+fi
+
+DOWNLOAD_URL="https://github.com/agentuity/cli/releases/download/v${VERSION}/${DOWNLOAD_FILENAME}"
+CHECKSUMS_URL="https://github.com/agentuity/cli/releases/download/v${VERSION}/checksums.txt"
+
+ohai "Downloading Agentuity CLI v${VERSION} for ${OS}/${ARCH}..."
+curl -L --progress-bar "$DOWNLOAD_URL" -o "$TMP_DIR/$DOWNLOAD_FILENAME" || abort "Failed to download from $DOWNLOAD_URL"
+
+if [[ "$OS" != "Windows" ]]; then
+  ohai "Downloading checksums for verification..."
+  if ! curl -L --silent "$CHECKSUMS_URL" -o "$TMP_DIR/checksums.txt"; then
+    warn "Failed to download checksums file. Skipping verification."
+  else
+    ohai "Verifying checksum..."
+    if command -v sha256sum >/dev/null 2>&1; then
+      CHECKSUM_TOOL="sha256sum"
+    elif command -v shasum >/dev/null 2>&1; then
+      CHECKSUM_TOOL="shasum -a 256"
+    else
+      warn "Neither sha256sum nor shasum found. Skipping checksum verification."
+      CHECKSUM_TOOL=""
+    fi
+    
+    if [[ -n "$CHECKSUM_TOOL" ]]; then
+      cd "$TMP_DIR"
+      COMPUTED_CHECKSUM=$($CHECKSUM_TOOL "$DOWNLOAD_FILENAME" | cut -d ' ' -f 1)
+      EXPECTED_CHECKSUM=$(grep "$DOWNLOAD_FILENAME" checksums.txt | cut -d ' ' -f 1)
+      
+      if [[ -z "$EXPECTED_CHECKSUM" ]]; then
+        warn "Checksum for $DOWNLOAD_FILENAME not found in checksums.txt. Skipping verification."
+      elif [[ "$COMPUTED_CHECKSUM" != "$EXPECTED_CHECKSUM" ]]; then
+        abort "Checksum verification failed. Expected: $EXPECTED_CHECKSUM, Got: $COMPUTED_CHECKSUM"
+      else
+        ohai "Checksum verification passed!"
+      fi
+      cd - > /dev/null
+    fi
+  fi
+fi
+
+ohai "Processing download..."
+if [[ "$OS" == "Windows" ]]; then
+  ohai "Downloaded Windows MSI installer to $TMP_DIR/$DOWNLOAD_FILENAME"
+  
+  if [[ -n "${CI}" || -n "${GITHUB_ACTIONS}" ]]; then
+    ohai "CI environment detected, skipping MSI installation and creating dummy executable"
+    mkdir -p "$INSTALL_PATH"
+    if [[ "$OS" == "Windows" ]]; then
```

Comment:
i dont think we should leak CI testing into the install
script. we need a better way to test on windows for
CI